### PR TITLE
[GStreamer] Do not send repeated encrypted events.

### PR DESCRIPTION
**Pull request by charlie-ht**
This will break YouTube tests, since if they generateRequest twice with
the same init data, the request is rejected in WebKit, and will return
DOM exceptions during the tests.

### CODE REVIEW

File: Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp [hunk 1]
```diff
@@ -40,7 +41,7 @@ struct _WebKitMediaCommonEncryptionDecryptPrivate {
     RefPtr<WebCore::CDMInstance> m_cdmInstance;
     WTF::HashMap<String, WebCore::InitData> m_initDatas;
     Vector<GRefPtr<GstEvent>> m_pendingProtectionEvents;
-    uint32_t m_currentEvent { 0 };
+    WTF::HashSet<WebCore::InitData> m_processedEventData;
```
Comment (calvaris) [inline Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp]: m_processedInitData

File: Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp [hunk 2]
```diff
@@ -347,61 +348,28 @@ static void webkitMediaCommonEncryptionDecryptProcessPendingProtectionEvents(Web
...
+        if (!mappedBuffer) {
+            GST_WARNING_OBJECT(self, "cannot map protection data");
             continue;
         }
+        WebCore::InitData initData = WebCore::InitData(mappedBuffer.data(), mappedBuffer.size());
 
-        if (priv->m_currentEvent == GST_EVENT_SEQNUM(event.get())) {
-            GST_TRACE_OBJECT(self, "event %u already handled", priv->m_currentEvent);
+        if (priv->m_processedEventData.contains(initData)) {
+            GST_TRACE_OBJECT(self, "event %u already handled", GST_EVENT_SEQNUM(event.get()));
```
Comment (calvaris) [inline Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp]: This message is misleading, it should say something like "event %u init data already handled"

**Review by calvaris — CHANGES_REQUESTED** [review]
Apart from the things I already commented, I suspect this creates decryption issues resulting in macroblocks.

**Comment by calvaris** [comment]
And more thorough testing, actually I would like to test them myself too.

**Comment by charlie-ht** [comment]
I can't test this currently with playready on the YTTV Sintel asset, because of bugs in OCDM server. The widevine assert creates macroblocks because we don't have a valid keybox for decryption there.

**Comment by charlie-ht** [comment]
Stale PR

**Pull request closed without merging**